src/core/Target_location.mli

```diff
@@ -0,0 +1,155 @@
+(* Cooper Pierce
+ *
+ * Copyright (C) 2023 Semgrep Inc.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * version 2.1 as published by the Free Software Foundation, with the
+ * special exception on linking described in file LICENSE.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the file
+ * LICENSE for more details.
+ *)
+
+(** Types for describing target locations, i.e., everything about a target
+    other than the actual contents.
```

Comment:
You have to explain what a target location is. It's not a standard term that we use elsewhere.

Reply:
How is it different from a source?

Reply:
I think I explain what it is (perhaps without comparison or sufficient differentiation) in the comment on `type t`.

I can add more elaboration on the difference with `Source.t` though. If the question here is more than rhetorical, the distinction I have here is that a "target location" comprises sufficient information to be able to create a target (e.g., what products should rules be run from, what is the analyzer type, is there an associated lockfile, etc...) whereas a "source" is just an origin for the bytes being scanned which supply the "contents" of a target.

Reply:
Please again do not use the term "location" which is used for a file range/region